src/MultiplayerMod/Multiplayer/CoreOperations/PlayersManagement/PlayersManagementController.cs

```diff
@@ -65,10 +69,12 @@ UnityTaskScheduler scheduler
     }
 
     private void OnClientStateChanged(MultiplayerClientState state) {
-        if (state != MultiplayerClientState.Connected)
-            return;
-
-        client.Send(new InitializeClientCommand(profileProvider.GetPlayerProfile()));
+        if (state == MultiplayerClientState.Connected)
+            client.Send(new InitializeClientCommand(profileProvider.GetPlayerProfile()));
+        if (state == MultiplayerClientState.Error) {
+            events.Dispatch(new StopMultiplayerEvent(multiplayer));
```

Comment:
Logically `StopMultiplayerEvent` is a decision of handling `ConnectionLostEvent`, but okay.

Reply:
Yep.
Lets leave it outside of this PR.
ConnectionLost also could be moved inside of PlayerStateChanged by adding another enum value since both things are quite similar.